src/poprox_concepts/domain/recommendation.py

```diff
@@ -0,0 +1,28 @@
+from typing import TypeAlias
+
+from pydantic import BaseModel, ConfigDict, Field
+
+from poprox_concepts.domain.article import Article
+
+PrimitiveTypes: TypeAlias = bool | int | float | str | bytes | None
+ValueTypes: TypeAlias = BaseModel | PrimitiveTypes
+
+CollectionTypes: TypeAlias = list[ValueTypes] | set[ValueTypes] | dict[ValueTypes, ValueTypes] | tuple[ValueTypes, ...]
+
+RecursiveCollectionTypes: TypeAlias = (
+    list[CollectionTypes] | set[CollectionTypes] | dict[ValueTypes, CollectionTypes] | tuple[CollectionTypes, ...]
+)
+
+SerializableTypes: TypeAlias = ValueTypes | CollectionTypes | RecursiveCollectionTypes
```

Comment:
How important are sets and tuples for this vs. just lists and dicts? If lists and dicts are sufficient, then this can just be `pydanticJsonValue`, can't it?

When these are round-tripped through JSON, sets and tuples will be parsed back as lists, so only the list and dict types will show up in deserialized objects.

Reply:
Yeah, that works! I was trying to be thorough but seems like I went a bit overboard 😅